library/manageiq_custom_attributes.py

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/python
+
+import os
+from ansible.module_utils.basic import *
+from miqclient.api import API as MiqApi
+
+
+DOCUMENTATION = '''
+---
+module: manageiq_custom_attributes
+short_description: add, update, delete an entity custom attributes in ManageIQ
+requirements: [ ManageIQ/manageiq-api-client-python ]
+author: Daniel Korn (@dkorn)
+options:
+  miq_url:
+    description:
+      - the manageiq environment url
+    default: MIQ_URL env var if set. otherwise, it is required to pass it
+  miq_username:
+    description:
+      - manageiq username
+    default: MIQ_USERNAME env var if set. otherwise, it is required to pass it
+  miq_password:
+    description:
+      - manageiq password
+    default: MIQ_PASSWORD env var if set. otherwise, it is required to pass it
+  entity_name:
+    description:
+      - the entity name in manageiq to which the custom attributes belongs
+    required: true
+    default: null
+  entity_type:
+    description:
+      - the entity type in manageiq to which the custom attributes belongs
+    required: true
+    default: null
+  state:
+    description:
+      - the state of the custom attributes
+      - On present, it will add the custom attributes to the entity if not
+      already exist, or update the custom attributes if the associated data
+      is different
+      - On absent, it will delete the custom attributes from the entity if exist
+    required: false
+    choices: ['present', 'absent']
+    default: 'present'
+  custom_attributes:
+    description:
+      - the custom attributes of the entity
+    required: true
+    default: null
+'''
+
+EXAMPLES = '''
+# Add custom attributes for Openshift Containers Provider in ManageIQ
+  manageiq_custom_attributes:
+    entity_type: 'provider'
+    entity_name: 'openshift01'
+    state: 'present'
+    miq_url: 'http://localhost:3000'
+    miq_username: 'admin'
+    miq_password: '******'
+    custom_attributes:
+      - name: "ca1"
+        value: "value 1"
+      - name: "ca2"
+        value: "value 2"
+'''
+
+
+class ManageIQ(object):
```

Comment:
Question: Why do all the main classes in the modules have the same name? Won't it be better to call it ManageIQCA or some other more specific name?